### PR TITLE
feat: add route group

### LINKED ISSUE #861: feat: Route group feature
# why
In order to allow users to filter out routes with a specified group from a large number of routes, I propose to add the route groups feature.

# how
## front-end
In order to finish this feature. There are 3 steps need to do.
- Adding group options to the edit or create route page.
image

- Route group list page
image

- Create or Edit route group page
image

## back-end
Create or update API to match front-end requirements.
- Create route and update route API need to update.
- Route Group List API.
- Create and update route group API.

# APISIX
Need to discuss how to implement route group requirements.

Free to comment here.

cc @juzhiyuan @liuxiran @gxthrj @nic-chen @membphis

### PR DESCRIPTION
**Pull request by LiteSun**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance

- Related issues

___
### Bugfix
- Description

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.

wait for related API ready.
resolves #861

### CODE REVIEW

**Comment by juzhiyuan** [comment]
CI pending, need to retrigger.

**Comment by LiteSun** [comment]
#1061 just merged, will access the API soon.

File: web/src/pages/Route/transform.ts [hunk 1]
```diff
@@ -35,8 +44,14 @@ export const transformStepData = ({
     };
   }
 
+  const labels = {};
+  transformLableValueToKeyValue(form1Data.labels).forEach(item => {
+    labels[item.labelKey] = item.labelValue;
```
Comment (LiteSun) [inline web/src/pages/Route/transform.ts:49]: if we have `version:v2` and `version:v3`, only the laster one will be posted, need to confirm. 
cc @juzhiyuan @nic-chen

Comment (liuxiran) [reply web/src/pages/Route/transform.ts:49]: both of them need to be posted

File: web/src/locales/en-US/component.ts [hunk 2]
```diff
@@ -32,6 +32,7 @@ export default {
   'component.global.loading': 'Loading',
   'component.global.list': 'List',
   'component.global.description': 'Description',
+  'component.global.label': 'Label',
```
Comment (liuxiran) [inline web/src/locales/en-US/component.ts:35]:
```suggestion
  'component.global.labels': 'Labels',
```

File: web/src/pages/Route/List.tsx [hunk 3]
```diff
@@ -90,6 +101,47 @@ const Page: React.FC = () => {
       dataIndex: 'desc',
       hideInSearch: true,
     },
+    {
+      title: formatMessage({ id: 'component.global.label' }),
+      dataIndex: 'labels',
+      render: (_, record) => {
+        return Object.keys(record.labels || {}).map((item) => (
+          <Tag key={Math.random().toString(36).slice(2)}>
+            {item}:{record.labels[item]}
```
Comment (liuxiran) [inline web/src/pages/Route/List.tsx:110]: just to make sure that if the `labels` we got from api is a `[key:string]:[value:string]` object

File: web/src/locales/zh-CN/component.ts [hunk 4]
```diff
@@ -32,6 +32,7 @@ export default {
   'component.global.loading': '加载中',
   'component.global.list': '列表',
   'component.global.description': '描述',
+  'component.global.label': '标签',
```
Comment (juzhiyuan) [inline web/src/locales/zh-CN/component.ts:35]:
```suggestion
  'component.global.labels': '标签',
```

File: web/src/pages/Route/List.tsx [hunk 5]
```diff
@@ -14,21 +14,32 @@
...
+
 const Page: React.FC = () => {
   const ref = useRef<ActionType>();
   const { formatMessage } = useIntl();
 
+  const [labelList, setLabelList] = useState<RouteModule.LabelList>({});
+
+  useEffect(() => {
+    fetchLabelList().then((data) => {
+      setLabelList(transformLabelList(data));
```
Comment (juzhiyuan) [inline web/src/pages/Route/List.tsx:40]: would better use a transformer in service, then you could set state by `then(setXXX)`

File: web/src/pages/Route/List.tsx [hunk 6]
```diff
@@ -90,6 +101,47 @@ const Page: React.FC = () => {
       dataIndex: 'desc',
       hideInSearch: true,
     },
+    {
+      title: formatMessage({ id: 'component.global.label' }),
+      dataIndex: 'labels',
+      render: (_, record) => {
+        return Object.keys(record.labels || {}).map((item) => (
+          <Tag key={Math.random().toString(36).slice(2)}>
```
Comment (juzhiyuan) [inline web/src/pages/Route/List.tsx:109]: why use this method to set key?

File: web/src/pages/Route/components/Step1/LabelsDrawer.tsx [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+  disabled,
+  labelsDataSource,
+  onClose,
+  onChange = () => { },
+}) => {
+  const transformLabel = transformLableValueToKeyValue(labelsDataSource);
+
+  const { formatMessage } = useIntl();
+  const [form] = Form.useForm();
+  const [labelList, setLabelList] = useState<RouteModule.LabelList>();
```
Comment (juzhiyuan) [inline web/src/pages/Route/components/Step1/LabelsDrawer.tsx:120]: default value?

**Comment by juzhiyuan** [comment]
e2e failed, @idbeta @nic-chen

File: web/src/locales/zh-CN/component.ts (hunk 4 above)
Comment (LiteSun) [reply web/src/locales/zh-CN/component.ts]: updated.

File: web/src/pages/Route/List.tsx (hunk 5 above)
Comment (LiteSun) [reply web/src/pages/Route/List.tsx]: > would better use a transformer in service, then you could set state by `then(setXXX)`

updated.

File: web/src/pages/Route/components/Step1/LabelsDrawer.tsx (hunk 7 above)
Comment (LiteSun) [reply web/src/pages/Route/components/Step1/LabelsDrawer.tsx]: > default value?

updated.

**Review by juzhiyuan — APPROVED** [review]
LGTM then.

**Review by liuxiran — APPROVED** [review]

**Comment by liuxiran** [comment]
> e2e failed, @idbeta @nic-chen

https://github.com/apache/apisix-dashboard/blob/e0af89e7df02de137aeac3d4bad53cf2efde41c6/api/test/e2e/consumer_test.go#L191

need a  `sleep` after this line

**Pull request merged**